code/game/objects/effects/traps.dm

```diff
@@ -264,7 +272,11 @@ Add those other swinging traps you mentioned above!
 	. = ..()
 	if(AM.is_incorporeal())
 		return
-	if (istype(AM, /mob/living))
+
+	if(!tripped)
```

Comment:
same as above
and you should probably add a proc that is "apply_damage(/mob/living/victim)" that is changed per trap, instead of overwriting Crossed() every time